crates/oci/src/client.rs

```diff
@@ -83,14 +92,11 @@ impl Client {
 
             let source = spin_trigger::parse_file_url(source.as_str())?;
             let layer = Self::wasm_layer(&source).await?;
-            let digest = &layer.sha256_digest();
-            layers.push(layer);
 
-            // Update the module source with the content digest of the layer.
-            c.source.content = ContentRef {
-                source: None,
-                digest: Some(digest.clone()),
-            };
+            // Update the module source with the content ref of the layer.
+            c.source.content = Self::content_ref_for_layer(&layer);
+
+            layers.push(layer);
```

Comment:
I didn't bother with skipping layers here because you _can't_ have a valid wasm file <=2 bytes and <=128 bytes is practically impossible too.